tools/c7n_slacker/c7n_slacker/sqs_handler.py

```diff
@@ -0,0 +1,348 @@
+                    self.logger.debug("Contact method is SNS topic. Skipping.")
+                    continue
+                else:
+                    self.logger.debug("ID number found. Doing LDAP lookup.")
+                    ldap_filter = '(%s=%s)' % (self.uid_key, resource_owner_value)
+                    connection.search(self.base_dn, ldap_filter, attributes=self.attributes)
+                    if len(connection.entries) == 0:
+                        self.logger.warning("User not found. base_dn: %s filter: %s", self.base_dn, ldap_filter)
+                        continue
+                    elif len(connection.entries) > 1:
+                        self.logger.warning("Multiple results returned.")
+                        continue
+                    else:
+                        resource_owner_value = connection.entries[0]['mail']
+
+            self.logger.debug("%s %s" % (resource_owner_value, resource_string))
+
+            resource_dict['resource_owner_value'] = resource_owner_value
+            resource_dict['resource_string'] = resource_string
+
+            for method in config.get('notify_methods'):
+                if method == 'Slack':
+                    slack = SlackBot(config.get('slack_token'), self.logger)
+                    slack.slack_handler(resource_dict)
+
+    def process_sqs(self, config):
+
+        sqs_fetch = sqsexec.MessageIterator(client=self.session.client('sqs'),
+                                            queue_url=self.config.get('queue_url'), timeout=0)
+        connection = self.get_ldap_session()
+        self.logger.info('Processing queue messages')
+
+        for m in sqs_fetch:
+
+            with ThreadPoolExecutor(max_workers=2) as w:
+                futures = {w.submit(self.message_handler, connection, config, m): m}
+
+                for future in as_completed(futures):
+                    # sqs_fetch.ack(m)
+                    if future.exception():
+                        self.logger.error("Error processing message: %s", future.exception())
+
+    @staticmethod
+    def session_factory(config):
+
+        if config.get('region') is None:
+            set_region = os.environ['AWS_DEFAULT_REGION']
+        else:
+            set_region = config.get('region')
+
+        if config.get('profile') is None:
+            set_profile = os.environ['AWS_DEFAULT_PROFILE']
+        else:
+            set_profile = config.get('profile')
+
+        return boto3.Session(
+            region_name=set_region,
+            profile_name=set_profile)
+
+    def get_resource_owner_values(self, sqs_message):
+        if 'Tags' in sqs_message:
+            tags = {tag['Key']: tag['Value'] for tag in sqs_message['Tags']}
+        else:
+            self.logger.debug("No tags found on resource. Skipping")
+            return None, None
+
+        if tags:
+            for contact_tag in self.config.get('contact_tags'):
+                for tag in tags:
+                    if tag == contact_tag:
```

Comment:
this is kind of a mess, you would be better off calling something like `tags.get(contact_tag, None)`